app/models/transaction.rb

```
@@ -0,0 +1,76 @@
+class Transaction < ActiveRecord::Base
+
+  belongs_to :portfolio
+  belongs_to :stock
+  # has_one :stock, foreign_key: "index_id"
+
+  def self.sell(qty, price, datetime, all_transactions)
+    return "Not enough stocks in portfolio" if (all_transactions.map(&:available_stock).reduce(:+)) < qty
```

This should probably raise an error instead of just return a string.